Sources/ComposableArchitecture/Store.swift

```diff
@@ -408,20 +411,28 @@ public final class Store<State, Action> {
     return self.scope(state: { $0 }, action: absurd)
   }
 
-  private enum ThreadCheckStatus {
+  private enum Event {
     case effectCompletion(Action)
     case `init`
     case scope
     case send(Action, originatingAction: Action?)
   }
+  
+  @inline(__always)
+  private func performChecks(event: Event) {
+    #if DEBUG
+    threadCheck(event: event)
+    stackCheck(event: event)
+    #endif
+  }
 
   @inline(__always)
```

Comment:
This could probably be removed if we make everything transit through `performChecks`